### ISSUE TITLE
(GitHub Issue Comments) The script no longer works: it shows invalid active items, no active items and the filters don't actually toggle anything

### ISSUE THREAD

**Issue by darkred** [Mottie/GitHub-userscripts#105]
Greetings @Mottie !

Using Chrome 78/Firefox 70 and only Tampermonkey beta 4.9.6088 and 'GitHub Issue Comments' v1.3.8.

After GitHub recently having changed [so many class names](https://github.com/Mottie/GitHub-userscripts/issues/104#issuecomment-552107392),
unfortunately the 'GitHub Issue Comments' script no longer works:

- it shows invalid active items when 'Toggle items' is disabled
- it shows no active items when 'Toggle items' is enabled
- the filters don't actually hide/show anything in any case.

There are no errors in Chrome Console.

I've tried various issue pages:

```
https://github.com/Mottie/GitHub-userscripts/issues/104
https://github.com/Mottie/GitHub-userscripts/issues/84
https://github.com/openstyles/stylus/issues/55
https://github.com/isaacs/github/issues/895
```
and in all cases, the above issues occur.

Also, the filters, in the rare cases when one is listed as active with 'Toggle items' enabled (e.g. in the first link its 'Reactions'),
don't actually toggle anything when clicked.

**Could it be possible to make it work again?? May I assist you in any way with this?**
_If I'm not mistaken, unfortunately it's not a simple matter of just tweaking the CSS selectors. 
( Maybe they should all be replaced with jQuery `:contains()` (specific string) selectors ?? )_

**Comment by Mottie** [comment]
Hey @darkred!

Hmm, yeah... it looks like a bunch of selectors have changed. Thanks GitHub!

I'll try to find some time soon to get this updated. It shouldn't be too bad (famous last words), so don't worry about assisting me. The userscript isn't using jQuery either 😸.

**Comment by darkred** [comment]
Hi there Mottie.

> The userscript isn't using jQuery either 😸.

I think I wasn't expressed properly, so let me rephrase please: 

I know that the userscript doesn't use jQuery:
it's just that, if I'm not mistaken, the CSS selectors of the various entry types to filter are no longer straightforward/descriptive, 
e.g. for "Label Changes" it's no longer something similar to `.discussion-item-labeled, .discussion-item-unlabeled` that you use in the script,
and so I'm afraid the only way to make the script work again is to use jQuery's `:contains()` (specific string), although I know that the `:contains` selector is much more expensive in terms of performance than the class selectors from before.

**Comment by Mottie** [comment]
Hey sorry for not updating this issue. I've been super busy! I got back from vacation at the end of the year, and it's been non-stop work (around the house & at work).

I did start working on this issue. I didn't encounter any problems with setting the script up without using jQuery. I'll see if I can get it done next week, since I'm going out of town tomorrow, for about a week.

**Comment by Mottie** [comment]
Sorry this took me so long to get fixed. Please test it out and let me know if you find any issues.

**Comment by darkred** [comment]
Thanks so much for fixing the script!! Awesome! Super!

---

I've noticed these issues:

- in this page, the "Reactions" filter doesn't become active even though I have two reaction emojis below your [second-to-last comment](https://github.com/Mottie/GitHub-userscripts/issues/105#issuecomment-577966939).
(The "Reactions" filter works ok in other pages I've tried)
- in PR pages, e.g. PRs in https://github.com/sindresorhus/refined-github/pulls,
the following filters don't work nor appear active:
  - `Commits`,
  - `Reviews`: 
    - _"x reviewed yesterday"_, 
    - _"x approved these changes 8 hours ago"_, 
    - _"x requested changes 3 days ago"_, 
    - _"x requested a review from y yesterday"_,
    - _"Resolved"/collapsed comments in PRs)_, 
  - `Merged` _(e.g. "x force-pushed the x branch from a to b yesterday")_
  - `Integrations`


- the issue I had reported in my last comment in https://github.com/Mottie/GitHub-userscripts/issues/84#issuecomment-504640028 still occurs (for that page): 
  > Τhe `+1 Comments` doesn't show up as active/available filter in the above issue page when enabling the global "Toggle items" filter, even though it has content.

**Comment by Mottie** [comment]
Hi @darkred!

Please check the reactions issue you mentioned above and in #84.

As for the PRs... ugh, somehow I thought you mean the list of PRs page. I'll take a look now.

**Comment by darkred** [comment]
> Please check the reactions issue you mentioned above and in #84.

I confirm both issues are fixed. Thank you!



> As for the PRs... ugh, somehow I thought you mean the list of PRs page.

Sorry, I should have been more descriptive about that issue.

**Comment by darkred** [comment]
Unfortunately, we both got confused!😵

You got confused about my report about PRs pages.

I now realized I got confused with your comment regarding #84! you wrote: 

> Please check the reactions issue you mentioned above and in #84.

but issue #84 is not also about `Reactions`: that's about the `+1 Comments` filter.
And that issue (still) occurs, as follows:

> Τhe `+1 Comments` doesn't show up as active/available filter in the above issue page when enabling the global "Toggle items" filter, even though it has content.


More details/STR:
- Go to page #84. In that page, you have a very short message ("Oops") https://github.com/Mottie/GitHub-userscripts/issues/84#issuecomment-495440160
- Initially (in a fresh installation), in the script's options, the toggle "Only show active items" is disabled default. So, if you click the `+1 Comments` filter (which is displayed as black/non-grey, therefore having content), you'll see that your very short message becomes filtered, as expected.
- If now, and that's the catch, if you enable the toggle "Only show active items", 
then your short comment is still filtered, as expected, **but the `+1 Comments` filter is no longer listed, although it should**. The same occurs if you reload the page: no `+1 Comments` anymore.



> Τhe `+1 Comments` doesn't show up as active/available filter in the above issue page when enabling the global "Toggle items" filter, even though it has content.

**Comment by Mottie** [comment]
Ok, I _think_ I fixed most of the issues; but I'm not happy with how the reviews are working.

Reviews are sometimes wrapped in an outer wrapper that includes a timeline comment with code comments. Sometimes the review wrapper includes other changes like commits and other timeline changes. 

So now there are new/updated review entries:

- `Reviews (All)` - hides _all_ reviews
- `Reviews (Outdated)` - hides only outdated reviews; but may include current reviews depending on the wrapper.
- `Reviews (Resolved)` - hides only resolved reviews, which usually means they are outdated. Also may include other reviews depending on the grouping
- `Reviews (Current)` - hides only current reviews; but again it may include other review types...

I also ended up removing the "Diffs (outdated) comments" entry since it has become redundant with "Review (outdated)".

**Comment by darkred** [comment]
Thank you for all your great work and effort! I'm grateful!

Regarding your concern on how the reviews are working, 
I can only say, that you have already done the best possible! 👍

---

The only issue that still exists is #84:
in the latest script version the `+1 Comments` filter:
- never appears to be active (i.e. that it has matching content), and 
- enabling it *(after disabling the `Only show active items` toggle)* doesn't hide any matching comments.

Example pages:
https://github.com/isaacs/github/issues/253
https://github.com/isaacs/github/issues/215
https://github.com/Mottie/GitHub-userscripts/issues/84  (your "Oops" comment)

**Comment by Mottie** [comment]
Ok, please check again. I misnamed the function, so hide +1 & reactions were both broken. Both should be fixed now. 🤞

**Comment by darkred** [comment]
Yes, I confirm it's fixed now. Closing both bugs now.
Thanks a lot for everything!
